### PR TITLE
feature(new integration): LiveChat web Device mode integration onboarding

### PR DESCRIPTION
**Pull request by RohitKhatta**
## Description of the change

> onboarding LiveChat
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/633)

### CODE REVIEW

File: integrations/LiveChat/browser.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  identify(rudderElement) {
+    logger.debug("===In LiveChat Identify===");
+    const { message } = rudderElement;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+
+    if (!email) {
+      logger.error("User parameter (email) is required for identify call");
+      return;
```
Comment (shrouti1507) [inline integrations/LiveChat/browser.js:90]: we do not drop any events from device mode. Have we checked what happens if the email is not set. Can we set name and other attributes without setting email?

File: integrations/LiveChat/browser.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+    window.LiveChatWidget.call("set_customer_email", email);
+
+    const name =
+      get(message, "context.traits.name") || get(message, "traits.name");
+
+    if (name) {
+      window.LiveChatWidget.call("set_customer_name", name);
+    }
+  }
```
Comment (shrouti1507) [inline integrations/LiveChat/browser.js:100]: what about the other session variables? We need to set them using message.traits. Please add that.

**Review by shrouti1507 — APPROVED** [review]

File: integrations/LiveChat/browser.js (hunk 2 above)
Comment (RohitKhatta) [reply integrations/LiveChat/browser.js:100]: added

File: integrations/LiveChat/browser.js (hunk 1 above)
Comment (RohitKhatta) [reply integrations/LiveChat/browser.js]: removed return type

File: integrations/LiveChat/browser.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    logger.debug("===In LiveChat Identify===");
+    const { message } = rudderElement;
+    const { traits } = rudderElement.message.context;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+
+    if (!email) {
+      logger.error("User parameter (email) is required for identify call");
+    }
+    window.LiveChatWidget.call("set_customer_email", email);
```
Comment (shrouti1507) [inline integrations/LiveChat/browser.js:93]:
```
if (email) {
      window.LiveChatWidget.call("set_customer_email", email);
    } else {
 logger.error("User parameter (email) ,required for identify call, not found.");
}
```

File: integrations/LiveChat/browser.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    const name =
+      get(message, "context.traits.name") || get(message, "traits.name");
+
+    if (name) {
+      window.LiveChatWidget.call("set_customer_name", name);
+    }
+    if (traits) {
```
Comment (shrouti1507) [inline integrations/LiveChat/browser.js:101]: also I see segment is adding userId as a field here. Please add it as well.

Comment (shrouti1507) [reply integrations/LiveChat/browser.js:101]: https://segment.com/docs/connections/destinations/catalog/livechat/#identify

File: integrations/LiveChat/browser.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    return !!window.LiveChatWidget;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In LiveChat Identify===");
+    const { message } = rudderElement;
+    const { traits } = rudderElement.message.context;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
```
Comment (shrouti1507) [inline integrations/LiveChat/browser.js:88]: As we are not having root traits for JS SDK, lets remove it. Keep it as
`get(message, "context.traits.email")` only

File: integrations/LiveChat/browser.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    if (!email) {
+      logger.error("User parameter (email) is required for identify call");
+    }
+    window.LiveChatWidget.call("set_customer_email", email);
+
+    const name =
+      get(message, "context.traits.name") || get(message, "traits.name");
```
Comment (shrouti1507) [inline integrations/LiveChat/browser.js:96]: same here. Lets remove root level traits

File: integrations/LiveChat/browser.js (hunk 5 above)
Comment (RohitKhatta) [reply integrations/LiveChat/browser.js:88]: removed root level traits

File: integrations/LiveChat/browser.js (hunk 3 above)
Comment (RohitKhatta) [reply integrations/LiveChat/browser.js:93]: resolved the flow

File: integrations/LiveChat/browser.js (hunk 4 above)
Comment (RohitKhatta) [reply integrations/LiveChat/browser.js:101]: added userId inside session variable

**Review by shrouti1507 — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**